src/components/Utils/orderResource.js

```diff
@@ -17,26 +15,33 @@ const saveOrder = (order, mutator) => {
   return method(order);
 };
 
+export const getFullOrderNumber = ({ poNumberPrefix = '', poNumberSuffix = '', poNumber = '' } = {}) => (
+  `${poNumberPrefix}${poNumber}${poNumberSuffix}`.trim()
+);
+
 export const updateOrderResource = (order, mutator, changedProps) => {
   const clonedOrder = cloneDeep(order);
 
+  clonedOrder.poNumber = getFullOrderNumber(clonedOrder);
+
   Object.assign(clonedOrder, changedProps);
 
   return saveOrder(clonedOrder, mutator);
 };
 
 export const createOrderResource = (order, mutator) => {
   const clonedOrder = cloneDeep(order);
-  const { numberPrefix = '', numberSuffix = '', poNumber = '' } = clonedOrder;
-  const fullOrderNumber = `${numberPrefix}${poNumber}${numberSuffix}`.trim();
 
-  clonedOrder.poNumber = fullOrderNumber || undefined;
+  clonedOrder.poNumber = getFullOrderNumber(clonedOrder) || undefined;
 
   return saveOrder(clonedOrder, mutator);
 };
 
 export const cloneOrder = (order, mutator, lines) => {
-  const clonedOrder = omit(order, ['id', 'adjustment', 'metadata', 'poNumber', 'workflowStatus', 'compositePoLines']);
+  const clonedOrder = omit(
+    order,
+    ['id', 'adjustment', 'metadata', 'poNumber', 'poNumberPrefix', 'poNumberSuffix', 'workflowStatus', 'compositePoLines'],
```

Comment:
What is the reason to skip here suffix and prefix?? To me it looks like a bug

Reply:
When user clones order PO number generates on BE side (only numbers). So no omitting pref/suf will brake PO number creation. I mean we'll get
`{
    poNumberPrefix: 'FOO',
    poNumber: '1001',
    poNumberSuffix: 'BAR'
}
`
instead of `{
    poNumberPrefix: 'FOO',
    poNumber: 'FOO1001BAR',
    poNumberSuffix: 'BAR'
}
`

Reply:
what makes you think like that? right now suffix and prefix are omitting and if you clone the order `{ poNumberPrefix: 'FOO', poNumber: '1001', poNumberSuffix: 'BAR' }` you'll get `{ poNumber: '1002' }`. So my question was why not to leave prefix and suffix, but now I see that PO will still be generated as `1002` and ui will be broken since it will try to remove suffix and prefix from that generated number